server/users/service.go

```diff
@@ -112,17 +123,17 @@ func (rs *RecognitionService) start() {
 			}
 
 			for destinationID, identifiers := range rp.DestinationsIdentifiers {
-				//recognized
-				if identifiers.UserID != "" {
-					err := rs.runPipeline(destinationID, identifiers)
+				if !identifiers.IsAllProperties() {
```

Comment:
Please try to avoid negative conditions, for instance this statement will be more clear:

```
if identifiers.IsAllProperties() {
  // Run pipeline only if all properties were recognized - it is needed to update all other anonymous events
  err = rs.runPipeline(destinationID, identifiers)
  if err != nil {
    logging.SystemErrorf("[%s] Error running recognizing pipeline: %v", destinationID, err)
  }
					
} else {
  // If some property is missing - event is anonymous
  err = rs.metaStorage.SaveAnonymousEvent(destinationID, identifiers.AnonymousID, identifiers.EventID, string(rp.EventBytes))
  if err != nil {
    logging.SystemErrorf("[%s] Error saving event with anonymous id %s: %v", destinationID, identifiers.AnonymousID, err)
  }
}
```

Reply:
Changed